crates/ide/src/rename.rs

```diff
@@ -1977,6 +1986,24 @@ impl Foo {
         );
     }
 
+    #[test]
+    fn test_self_outside_of_methods() {
+        cov_mark::check!(rename_self_outside_of_methods);
+        check(
+            "foo",
+            r#"
+fn f($0self) -> i32 {
+    self.i
```

Comment:
can we put a `use self as _;` into the body here? Just to check that we aren't accidentally touching the `self` modules